src/main/kotlin/com/exactpro/th2/check1/rule/sequence/SilenceCheckTask.kt

```diff
@@ -43,19 +43,26 @@ class SilenceCheckTask(
     messageStream: Observable<StreamContainer>,
     eventBatchRouter: MessageRouter<EventBatch>
 ) : AbstractCheckTask(ruleConfiguration, submitTime, sessionKey, parentEventID, messageStream, eventBatchRouter) {
-    private var protoPreMessageFilter: RootMessageFilter? = protoPreFilter.toRootMessageFilter()
-    private var messagePreFilter: SailfishFilter? = SailfishFilter(
-        CONVERTER.fromProtoPreFilter(protoPreMessageFilter!!),
-        protoPreMessageFilter!!.toCompareSettings()
-    )
-    private var metadataPreFilter: SailfishFilter? = protoPreMessageFilter!!.metadataFilterOrNull()?.let {
-        SailfishFilter(
-            CONVERTER.fromMetadataFilter(it, VerificationUtil.METADATA_MESSAGE_NAME),
-            it.toComparisonSettings()
+
+    private inner class Refs(protoPreFilter: PreFilter) {
```

Comment:
Can we make it static? Maybe all of the functions we use can be moved outside the instance of AbstractCheckTask